trade-finance-manager-api/api-tests/v1/amendments/create-amendment-tfm-object-on-amendment-completion.api-test.js

```diff
@@ -52,89 +63,177 @@ describe('update amendment-tfm on amendment completion', () => {
-    externalApis.getFacilityExposurePeriod = jest.fn(() => Promise.resolve({ exposurePeriodInMonths: 5 }));
-
-    const result = await amendmentController.createAmendmentTFMObject(mockAmendment.amendmentId, mockAmendment.facilityId);
-
-    const expected = {
-      amendmentExposurePeriodInMonths: 5,
-      coverEndDate: unixTime,
-      exposure: {
-        exposure: '4,000.00',
-        timestamp: expect.any(Number),
-        ukefExposureValue: 4000,
-      },
-      value: {
-        currency: CURRENCY.GBP,
-        value: 5000,
-      },
-    };
-
-    expect(result).toEqual(expected);
+  describe('TFM Facility end date feature flag disabled', () => {
+    beforeEach(() => {
+      jest.mocked(isTfmFacilityEndDateFeatureFlagEnabled).mockReturnValue(false);
+    });
+
+    it('createAmendmentTFMObject() - should not create tfm object when no getLatestCompletedAmendmentValue or getLatestCompletedAmendmentDate', async () => {
+      externalApis.getLatestCompletedAmendmentValue = jest.fn(() => Promise.resolve({}));
+      externalApis.getLatestCompletedAmendmentDate = jest.fn(() => Promise.resolve({}));
+      const result = await amendmentController.createAmendmentTFMObject(mockAmendment.amendmentId, mockAmendment.facilityId);
+
+      expect(result).toEqual({});
+    });
+
+    it('createAmendmentTFMObject() - should create tfm object when getLatestCompletedAmendmentValue and getLatestCompletedAmendmentDate', async () => {
+      externalApis.getFacilityExposurePeriod = jest.fn(() => Promise.resolve({ exposurePeriodInMonths: 5 }));
+
+      const result = await amendmentController.createAmendmentTFMObject(mockAmendment.amendmentId, mockAmendment.facilityId);
+
+      const expected = {
+        amendmentExposurePeriodInMonths: 5,
+        coverEndDate: unixTime,
+        exposure: {
+          exposure: '4,000.00',
+          timestamp: expect.any(Number),
+          ukefExposureValue: 4000,
+        },
+        value: {
+          currency: CURRENCY.GBP,
+          value: 5000,
+        },
+      };
+
+      expect(result).toEqual(expected);
+    });
+
+    it('createAmendmentTFMObject() - should create tfm object when getLatestCompletedAmendmentValue only', async () => {
+      externalApis.getLatestCompletedAmendmentDate = jest.fn(() => Promise.resolve({}));
+
+      const result = await amendmentController.createAmendmentTFMObject(mockAmendment.amendmentId, mockAmendment.facilityId);
+
+      const expected = {
+        exposure: {
+          exposure: '4,000.00',
+          timestamp: expect.any(Number),
+          ukefExposureValue: 4000,
+        },
+        value: {
+          currency: CURRENCY.GBP,
+          value: 5000,
+        },
+      };
+
+      expect(result).toEqual(expected);
+    });
+
+    it('createAmendmentTFMObject() - should create tfm object when only getLatestCompletedAmendmentDate', async () => {
```

Comment:
Just to match some of the other tests we have, do you think it would be worth putting these tests in a describe block like the below:
```typescript
describe('createAmendmentTfmObject', () => {
  it('should create tfm object when only getLatestCompletedAmendmentDate', ...);
});
```

Perfectly happy for you to not do this change though considering that you are updating tests rather than completely rewriting them (and because you've already made nice changes in other test files)